x-pack/plugins/endpoint/server/services/endpoint/alert_query_builders.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import qs from 'querystring';
+import { KibanaRequest } from 'kibana/server';
+import { EndpointAppConstants } from '../../../common/types';
+import { EndpointAppContext } from '../../types';
+
+export const kibanaRequestToAlertListQuery = async (
+  request: KibanaRequest<any, any, any>,
+  endpointAppContext: EndpointAppContext
+): Promise<Record<string, any>> => {
+  const pagingProperties = await getPagingProperties(request, endpointAppContext);
+  return {
+    body: {
+      query: {
+        match_all: {},
+      },
+      sort: [
+        {
+          '@timestamp': {
+            order: 'desc',
+          },
+        },
+      ],
+    },
+    from: pagingProperties.pageIndex * pagingProperties.pageSize,
+    size: pagingProperties.pageSize,
+    index: EndpointAppConstants.ALERT_INDEX_NAME,
+  };
+};
+
+async function getPagingProperties(
+  request: KibanaRequest<any, any, any>,
+  endpointAppContext: EndpointAppContext
+) {
+  const config = await endpointAppContext.config();
+  let pagingProperties: { page_size?: number; page_index?: number } = {};
+
+  if (request?.route?.method === 'get') {
+    if (typeof request?.url?.query === 'string') {
+      const qp = qs.parse(request.url.query);
```

Comment:
Does this obviate the schema validation phase? It might be better to have a slight bug in param order vs insecure http handlers. 

Also, `Number` isn't necessarily going to return integers.